src/source-maps.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2017 WebAssembly Community Group participants
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef WABT_SOURCE_MAPS_H
+#define WABT_SOURCE_MAPS_H
+#include <cstdint>
+#include <cstring>
+#include <map>
+#include <string>
+#include <vector>
+
+struct SourceMap {
+  static constexpr const int32_t kSourceMapVersion = 3;
+  // Representation of mappings
+  struct Segment {
+    // Field 1
+    uint32_t generated_col = 0;  // Start column in generated code. Remove?
```

Comment:
Yeah, seems like these aren't really necessary since you can always calculate them. Maybe better to just add functions if they're needed?